OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -225,6 +225,11 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Joint, ModelComponent);
     // Utility
     bool isCoordinateUsed(const Coordinate& aCoordinate) const;
 
+    /** Add a frame to the *frames* property in this Joint. The frame is
+     * adopted, and should have been dynamically allocated.
+     * Use this function instead of append_frames(). */
+    void addFrame(PhysicalOffsetFrame* frame);
+
```

Comment:
@chrisdembia  Was this necessary for this PR?

Reply:
Yes. There was coding that was using `append_frames()` to copy components into the `Joint`. and this was causing connections to fail because the connection was established with the original component outside the model, not with the component in the model.